internal/test/cleanup/cleanup.go

```diff
@@ -99,12 +99,17 @@ func CleanUpCloudstackTestResources(ctx context.Context, clusterName string, dry
 	if err != nil {
 		return fmt.Errorf("building cmk executable: %v", err)
 	}
-	cmk := executableBuilder.BuildCmkExecutable(tmpWriter, *execConfig)
-	defer cmk.Close(ctx)
+	for _, config := range execConfig.Profiles {
+		cmk := executableBuilder.BuildCmkExecutable(tmpWriter, config)
+		defer cmk.Close(ctx)
 
-	if err := cmk.ValidateCloudStackConnection(ctx); err != nil {
-		return fmt.Errorf("validating cloudstack connection with cloudmonkey: %v", err)
-	}
+		if err := cmk.ValidateCloudStackConnection(ctx); err != nil {
+			return fmt.Errorf("validating cloudstack connection with cloudmonkey: %v", err)
```

Comment:
I think it may be better to capture these errors in a multierror instead of returning in the middle of this loop. If the cloudstack connection fails for a profile, I'd `continue` to the next one

Reply:
and then we can close on the `continue` as well as at the end of each loop